### ISSUE TITLE
Kafka source vertex is unable to consume messages due to mismatched body size

### ISSUE THREAD

**Issue by louisnow** [numaproj/numaflow#749]
**Describe the bug**
The Kafka source vertex is unable to proceed ahead and consume messages.

```
2023-05-25T05:28:46.732Z	ERROR	numaflow.Source-processor	forward/forward.go:527	Retrying failed messages	{"vertex": "atlan-nf-pipe-webhooks-kafka-source", "errors": {"expected to write body size of -10180 but got 55356":1,"expected to write body size of -13472 but got 117600":1,"expected to write body size of -14021 but got 51515":1,"expected to write body size of -19079 but got 46457":1,"expected to write body size of -19294 but got 46242":1,"expected to write body size of -20173 but got 45363":1,"expected to write body size of -21051 but got 44485":1,"expected to write body size of -23530 but got 42006":1,"expected to write body size of -24700 but got 40836":1,"expected to write body size of -24837 but got 106235":1,"expected to write body size of -28169 but got 37367":1,"expected to write body size of 18998 but got 150070":1,"expected to write body size of 7229 but got 72765":1}, "pipeline": "atlan-nf-pipe-webhooks", "vertex": "kafka-source", "buffer": "default-atlan-nf-pipe-webhooks-kafka-source-filter"}
github.com/numaproj/numaflow/pkg/forward.(*InterStepDataForward).writeToBuffer
	/home/runner/work/numaflow/numaflow/pkg/forward/forward.go:527
github.com/numaproj/numaflow/pkg/forward.(*InterStepDataForward).writeToBuffers
	/home/runner/work/numaflow/numaflow/pkg/forward/forward.go:475
github.com/numaproj/numaflow/pkg/forward.(*InterStepDataForward).forwardAChunk
	/home/runner/work/numaflow/numaflow/pkg/forward/forward.go:353
github.com/numaproj/numaflow/pkg/forward.(*InterStepDataForward).Start.func1
	/home/runner/work/numaflow/numaflow/pkg/forward/forward.go:146
```

**To Reproduce**

1. We aren't sure how we can trigger it, deleting the pipeline and recreating it does not help.
2. Taking down the pipeline and resetting the consumer group fixes it but isn't a feasible solution

**Expected behavior**
The source vertex should be able to consume the message form Kafka

**Screenshots**

**Environment (please complete the following information):**

- Kubernetes: 1.24
- Numaflow: 0.8.0

**Additional context**

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
@jy4096 - what is the reason we use `int16` in [serde.go](https://github.com/numaproj/numaflow/blob/main/pkg/isb/serde.go#L184-L185)?

**Comment by jy4096** [comment]
> @jy4096 - what is the reason we use `int16` in [serde.go](https://github.com/numaproj/numaflow/blob/main/pkg/isb/serde.go#L184-L185)?

No particular reason. Thought int16 might be enough.

**Comment by louisnow** [comment]
Thanks @whynowy for the quick resolution
Is it possible to backport this fix to 0.8 instead of waiting till 0.9 comes out? We don't have any real workaround for this bug right now

**Comment by vigith** [comment]
We should release 0.8.1
